crates/rolldown/src/module_loader/module_loader.rs

```diff
@@ -24,13 +27,19 @@ use crate::{SharedOptions, SharedResolver};
 
 pub struct IntermediateNormalModules {
   pub modules: IndexVec<NormalModuleId, Option<NormalModule>>,
+  pub module_side_effects: IndexVec<NormalModuleId, Option<Arc<PackageJson>>>,
```

Comment:
We probably don't need this to get package.json.

Reply:
OK, is there another place we already store the related `PackageJson`, I can't find the related logic in our codebase.

Reply:
The code is placed in resolver. Let me create a PR to show it. I'm almost done it.

Reply:
https://github.com/rolldown/rolldown/pull/1170